syntax = "proto3";

package todo;

message Todo {
  uint64 id = 1;
  string name = 2;
  bool completed = 3;
  uint64 day = 4;
}

message CreateTodoRequest {
  string name = 1;
  bool completed = 2;
  uint64 day = 3;
}

message CreateTodoResponse {
  Todo todo = 1;
}

message ReadTodoRequest {
  uint64 id = 1;
}

message ReadTodoResponse {
  Todo todo = 1;
}

message ListTodosRequest {}

message ListTodosResponse {
  repeated Todo todos = 1;
}


service TodoService {
  rpc CreateTodo(CreateTodoRequest) returns (CreateTodoResponse);
  rpc ListTodos(ListTodosRequest) returns (ListTodosResponse);
  rpc ReadTodo(ReadTodoRequest) returns (ReadTodoResponse);
  }
